source/core/options.ts

```diff
@@ -847,12 +847,12 @@ const cloneInternals = (internals: typeof defaultInternals) => {
 		},
 		timeout: {...internals.timeout},
 		hooks: {
-			init: [...hooks.init],
-			beforeRequest: [...hooks.beforeRequest],
-			beforeError: [...hooks.beforeError],
-			beforeRedirect: [...hooks.beforeRedirect],
-			beforeRetry: [...hooks.beforeRetry],
-			afterResponse: [...hooks.afterResponse],
+			init: is.array(hooks.init) ? [...hooks.init] : hooks.init,
```

Comment:
I think it's better to use the native methods when available. `Array.isArray()`.

Reply:
I took a look at `is.array`. Is preference to `Array.isArray()` because we don't need logic related to `assertion` argument?

![Screenshot 2022-06-06 at 7 42 55 PM](https://user-images.githubusercontent.com/46992350/172156301-6a47a733-ec42-4f17-802a-2d34e6847580.png)